course_discovery/apps/api/clients/enterprise_catalog.py

```diff
@@ -0,0 +1,122 @@
+        return params
+
+    def get_content_metadata(self, catalog_uuid, query_params=None):
+        url = self._build_endpoint_url(catalog_uuid)
+        merged_params = self._merge_query_params(query_params)
+
+        logger.info(
+            'Fetching content metadata from Enterprise Catalog service. UUID: %s',
+            catalog_uuid,
+        )
+
+        try:
+            response = self.api_client.get(url, params=merged_params, timeout=self.REQUEST_TIMEOUT)
+            response_data = response.json()
+
+            logger.info(
+                'Successfully fetched enterprise catalog content. UUID: %s, Count: %s',
+                catalog_uuid,
+                response_data.get('count', 0),
+            )
+
+            return response_data
+
+        except HttpClientError as exc:
+            status_code = exc.response.status_code if hasattr(exc, 'response') else None
+
+            if status_code == 404:
+                logger.warning('Enterprise catalog not found. UUID: %s', catalog_uuid)
+                raise EnterpriseCatalogNotFoundError(
+                    status_code=404,
+                    response_data=exc.response.json() if hasattr(exc, 'response') and exc.response.content else None,
+                ) from exc
+
+            if status_code in (401, 403):
+                logger.error(
+                    'Service-to-service authentication failed for Enterprise Catalog API. Status: %s',
+                    status_code,
+                )
+                raise EnterpriseCatalogAuthError(
+                    status_code=status_code,
+                    response_data=exc.response.json() if hasattr(exc, 'response') and exc.response.content else None,
+                ) from exc
+
+            logger.error(
+                'Client error fetching content from Enterprise Catalog. UUID: %s, Status: %s',
+                catalog_uuid,
+                status_code,
+            )
+            raise EnterpriseCatalogAPIError(
+                status_code=status_code,
+                response_data=exc.response.json() if hasattr(exc, 'response') and exc.response.content else None,
+            ) from exc
+
+        except HttpServerError as exc:
+            status_code = exc.response.status_code if hasattr(exc, 'response') else None
+            logger.error('Server error from Enterprise Catalog. UUID: %s, Status: %s', catalog_uuid, status_code)
+            raise EnterpriseCatalogAPIError(
+                status_code=status_code,
+                response_data=exc.response.json() if hasattr(exc, 'response') and exc.response.content else None,
+            ) from exc
+
+        except requests.exceptions.Timeout as exc:
+            logger.error('Timeout fetching content from Enterprise Catalog. UUID: %s', catalog_uuid)
+            raise EnterpriseCatalogTimeoutError(
+                response_data={'catalog_uuid': catalog_uuid},
+            ) from exc
+
+        except requests.exceptions.ConnectionError as exc:
```

Comment:
We don't have any approach to handle this exception different than the others, we just fail and that's it, so instead of caching this particular exception, please use the base Exception from Requests: https://docs.python-requests.org/en/latest/api/#requests.RequestException to catch all request errors. @daviidco